### PR TITLE
fix(AntagSelectionSystem): Only create a new mind for ghosts

### LINKED ISSUE #30700: All syndicate are now marked as having an (Unknown) job in the ghost warp menu
## Description

This is EXTREMELY ABUSEABLE and meta-gameable.

**Reproduction**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by Brandon-Huu**
## About the PR

Made it so ``MakeAntag()`` only creates a new mind when transferring between entities
or sessions without a mind.

## Why / Balance

Resolves #30700 

## Technical details

~~Are all people who take ghost roles guaranteed to have the GhostComponent, if not, this partially reintroduces the bugs that were fixed in the previous pr~~

## Media

# Before
Screenshot from 2024-08-06 07-42-21

# After
Screenshot from 2024-08-06 07-40-04

## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by Brandon-Huu** [comment]
Draft so I can test my changes and let people know I'm working on it to avoid duplication of work 🫡

**Comment by Brandon-Huu** [comment]
So uuh

File: Content.Server/Antag/AntagSelectionSystem.cs [hunk 1]
```diff
@@ -341,12 +341,17 @@ public void MakeAntag(Entity<AntagSelectionComponent> ent, ICommonSession? sessi
...
+            var curMind = session.GetMind(); 
 
-            _mind.TransferTo(curMind, antagEnt, ghostCheckOverride: true);
-            _role.MindAddRoles(curMind, def.MindComponents, null, true);
-            ent.Comp.SelectedMinds.Add((curMind, Name(player)));
+            if (curMind == null || HasComp<GhostComponent>(session.AttachedEntity))
+            {
+                curMind = _mind.CreateMind(session.UserId, Name(antagEnt.Value));
+                _mind.SetUserId(curMind.Value, session.UserId);
+            }
```
Comment (EmoGarbage404) [inline Content.Server/Antag/AntagSelectionSystem.cs:350]: this should probably be checking if the new antag entity is different from the player's original entity in addition to the null check.

that would cover basically all the times an antagonist changes your entity and requires a new mind to be made.

Comment (Brandon-Huu) [reply Content.Server/Antag/AntagSelectionSystem.cs]: Changed it.
I originally intended to do this but didn't for whatever reason.

**Comment by Brandon-Huu** [comment]
UUUH

**Comment by Brandon-Huu** [comment]
It probably shouldn't create a new mind on ghostroles that aren't spawners?
Do ghostroles that don't have players connected to them possess a mind?
Could a player not just take over that mind?

**Comment by Errant-4** [comment]
Oh cool, this also coincidentally solves the bug where AutoTraitor infinitely loops creating a new mind and crashes Debug

**Comment by Brandon-Huu** [comment]
> > It probably shouldn't create a new mind on ghostroles that aren't spawners? Do ghost roles that don’t have players connected to them, possess a mind? If so, Could a player not just take over that mind?
> > I'm guessing I can do that with `_mind.SetUserId` ?
> 
> Doing this might fix the issue with the reinforcement teleporter?

[Uploading Screencast from 08-06-2024 10:07:16 AM.webm…]()

**Comment by Brandon-Huu** [comment]
Tired. Taking a break from coding for a while. someone else can deal with this.

**Pull request closed without merging**